File: gateway/gateway/src/main/java/es/jaime/gateway/orders/_shared/domain/Order.java
package es.jaime.gateway.orders._shared.domain;

import es.jaime.gateway._shared.domain.Aggregate;
import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
public abstract class Order extends Aggregate {
    @Getter protected OrderId orderId;
    @Getter protected OrderClientId clientId;
    @Getter protected OrderDate date;
    @Getter protected OrderType type;
    @Getter protected OrderState state;
    @Getter protected OrderTicker ticker;
}

File: client-frontend/src/model/orders/Order.ts
export interface Order {
    orderId: string;
    clientId: string;
    date: string;
    pendingOrderType: PendingOrderType;
    state: OrderState;
    ticker: string;
}

export enum OrderState {
    PENDING,
    CANCELLED,
    ERROR,
    EXECUTED,
}

export enum PendingOrderType {
    BUY,
    SELL,
    CANCEL,
}